### ISSUE TITLE
Internal server error - 365 Online not working

### ISSUE THREAD

**Issue by iahoeu** [ksainc/integration_ews#25]
**Describe the bug**
Exchange 365 Online not working - Internal server error

**To Reproduce**
1. Go to 'Connected Accounts'
2. Change Provider to Microsoft Exchange 365 Online
3. Click on Conenct
4. Popup of Internal server error

**Log**
Exception: OCA\EWS\Controller\UserConfigurationController::ConnectMS365(): Argument #1 ($code) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 230 in file '/var/www/nextcloud/apps/integration_ews/lib/Controller/UserConfigurationController.php' line 118
/var/www/nextcloud/lib/private/AppFramework/App.php - line 183:

OC\AppFramework\Http\Dispatcher->dispatch()

/var/www/nextcloud/lib/private/Route/Router.php - line 315:

OC\AppFramework\App::main()

/var/www/nextcloud/lib/base.php - line 1068:

OC\Route\Router->match()

/var/www/nextcloud/index.php - line 36:

OC::handleRequest()

Čím způsobeno TypeError: OCA\EWS\Controller\UserConfigurationController::ConnectMS365(): Argument #1 ($code) must be of type string, null given, called in /var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php on line 230
/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 230:

OCA\EWS\Controller\UserConfigurationController->ConnectMS365()

/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php - line 137:

OC\AppFramework\Http\Dispatcher->executeController()

/var/www/nextcloud/lib/private/AppFramework/App.php - line 183:

OC\AppFramework\Http\Dispatcher->dispatch()

/var/www/nextcloud/lib/private/Route/Router.php - line 315:

OC\AppFramework\App::main()

/var/www/nextcloud/lib/base.php - line 1068:

OC\Route\Router->match()

/var/www/nextcloud/index.php - line 36:

OC::handleRequest()

**Comment by ksainc** [comment]
Hi evilphish,

You only need to select single tenant, the important part is are redirect in the authentication and the api permissions.

I will make a how to guide and post it. Give me a day or so.

Sebastian

**Comment by arno1979** [comment]
I am also getting this problem. I followed the guide provided under the link you've given and I am also familiar with the process of registering an app in the Microsoft Admin Center as I have done it successfully for another app (also used in a Nextcloud installation).
I think everything is setup correctly on both sides: I noticed that when I removed the redirect URL in the MS admin panel the process instantly fails with a message from Microsoft (within the pop-up window displaying the login / authentication form).
I would love to get an update on your guide specific for the EWS Connector!
